lib/portage/sync/syncbase.py

```diff
@@ -252,6 +252,15 @@ def _refresh_keys(self, openpgp_env):
 		"""
 		out = portage.output.EOutput(quiet=('--quiet' in self.options['emerge_config'].opts))
 
+		# Set global proxy setting to Gemato ephemeral env
+		if 'http_proxy' in os.environ:
+			openpgp_env.proxy = os.environ['http_proxy']
```

Comment:
This isn't guaranteed to work. You need to pass `proxy` to the constructor, otherwise `dirmngr` may be started earlier and without your proxy.

Reply:
I'll adjust in the constructor then. Even if each tests I've ran are working maybe some configurations will not.